tests/tracking/test_tracking.py

```diff
@@ -169,12 +169,16 @@ def test_start_run_context_manager():
         assert persisted_run.info == first_run.info
     finished_run = tracking.MlflowClient().get_run(first_uuid)
     assert finished_run.info.status == RunStatus.to_string(RunStatus.FINISHED)
+
     # Launch a separate run that fails, verify the run status is FAILED and the run UUID is
     # different
-    with pytest.raises(Exception, match="Failing run!"):
+    try:
         with start_run() as second_run:
             second_run_id = second_run.info.run_id
             raise Exception("Failing run!")
+    except Exception:
+        pass
```

Comment:
`with pytest.raises` is not equivalent to `try-catch`. The new code doesn't throw if `with start_run()` swallows the error, which is not the desired behavior here.